### PR TITLE
Rename last XRP Classes

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

In JavaScript, we prefer `Xrp` to `XRP`. 

### Context of Change

This PR rounds out the last of changes for this rename. 

Unfortunately, there's no great way to use one internal class, so several internal classes needed to be split as well. This is super messy right now, but will get better as soon as we remove the deprecated classes in a few weeks. 

This change is non-breaking. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Some classes are deprecated. 

## Test Plan

CI

### CODE REVIEW

File: src/XRP/reliable-submission-xrp-client.ts [hunk 1]
```diff
@@ -19,6 +19,140 @@ async function sleep(milliseconds: number): Promise<void> {
...
+    const { lastLedgerSequence } = rawTransactionStatus
+    if (!lastLedgerSequence) {
+      return Promise.reject(
+        new Error(
+          'The transaction did not have a lastLedgerSequence field so transaction status cannot be reliably determined.',
+        ),
+      )
+    }
+
+    // Decode the sending address to a classic address for use in determining the last ledger sequence.
```
Comment (austin-king) [inline src/XRP/reliable-submission-xrp-client.ts:72]: nit: but should we shift these to `/* */` style comments as opposed to a new `//` for each line?

File: src/XRP/reliable-submission-xrp-client.ts [hunk 2]
```diff
@@ -19,6 +19,140 @@ async function sleep(milliseconds: number): Promise<void> {
...
+    // Retrieve the latest ledger index.
+    let latestLedgerSequence = await this.getLatestValidatedLedgerSequence(
+      sourceClassicAddress,
+    )
+
+    // Poll until the transaction is validated, or until the lastLedgerSequence has been passed.
+    /*
+     * In general, performing an await as part of each operation is an indication that the program is not taking full advantage of the parallelization benefits of async/await.
+     * Usually, the code should be refactored to create all the promises at once, then get access to the results using Promise.all(). Otherwise, each successive operation will not start until the previous one has completed.
+     * But here specifically, it is reasonable to await in a loop, because we need to wait for the ledger, and there is no good way to refactor this.
```
Comment (austin-king) [inline src/XRP/reliable-submission-xrp-client.ts:100]: helpful comment

File: src/XRP/reliable-submission-xrp-client.ts (hunk 1 above)
Comment (keefertaylor) [reply src/XRP/reliable-submission-xrp-client.ts:72]: I tend to prefer `//` inside of functions.

[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/XRP/reliable-submission-xrp-client.ts:72]: I think it is: https://xrpl.org/rippleapi-reference.html#classic-address

Happy to hear some details about what you'd rather say here.

File: src/XRP/reliable-submission-xrp-client.ts [hunk 3]
```diff
@@ -19,6 +19,140 @@ async function sleep(milliseconds: number): Promise<void> {
...
+          'The transaction did not have a lastLedgerSequence field so transaction status cannot be reliably determined.',
+        ),
+      )
+    }
+
+    // Decode the sending address to a classic address for use in determining the last ledger sequence.
+    // An invariant of `getLatestValidatedLedgerSequence` is that the given input address (1) exists when the method
+    // is called and (2) is in a classic address form.
+    //
+    // The sending address should always exist, except in the case where it is deleted. A deletion would supersede the
```
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/XRP/reliable-submission-xrp-client.ts:76]: I'm confused on what you mean.

> Does this mean that you can start a transaction with a deleted sending address
You can't start a transaction with a deleted address. 

> or that the sending address is deleted during the course of the transaction

There's a chance that you could send a payment and an account delete at the same time.

**Pull request merged**